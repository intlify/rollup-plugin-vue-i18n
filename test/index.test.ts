import { rollup } from 'rollup'
import VuePlugin from 'rollup-plugin-vue'
import JSONPlugin from '@rollup/plugin-json'
import YAMLPlugin from '@rollup/plugin-yaml'
import I18nPlugin from '../src/index'
import { pluginInline as InlinePlugin } from './utils'

async function setupRollup (options: any = {}) {
  const i18nLang = options.lang || 'json'
  const i18nLocaleMessages = options.localeMessages || `
{
  "en": {
    "hello": "hello!"
  },
  "ja": {
    "hello": "こんにちは！"
  }
}
`
  return rollup({
    input: '/entry.vue',
    plugins: [
      InlinePlugin(
        '/entry.vue',
        `
<template>
  <p>{{ $t('hello') }}</p>
</template>
<script>
export default {
  name: 'Entry'
}
</script>
<i18n lang="${i18nLang}">
${i18nLocaleMessages}
</i18n>
`
      ),
      JSONPlugin(),
      YAMLPlugin(),
      I18nPlugin(),
      VuePlugin({
        customBlocks: ['i18n']
      })
    ]
  })
    .then(bundle => bundle.generate({ format: 'es' }))
    .then(generated => generated.output[0])
}

const EXPECT_INJECT_STRING = `const options = typeof Component === 'function' ? Component.options : Component`

test('basic', async () => {
  const { code } = await setupRollup()

  expect(code).toMatchSnapshot()
  expect(code).toEqual(expect.stringContaining(EXPECT_INJECT_STRING))
})

test('yaml', async () => {
  const { code } = await setupRollup({
    lang: 'yaml',
    localeMessages: `
en:
  hello: "hello!"
ja:
  hello: "こんにちは！"
`
  })

  expect(code).toMatchSnapshot()
  expect(code).toEqual(expect.stringContaining(EXPECT_INJECT_STRING))
})
