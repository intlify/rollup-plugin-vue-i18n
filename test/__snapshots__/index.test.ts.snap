// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"import { normalizeComponent } from 'vue-runtime-helpers';

//
//
//
//

var script = {
  name: 'Entry'
};

var en = {
	hello: \\"hello!\\"
};
var ja = {
	hello: \\"こんにちは！\\"
};
const __i18n = {
	en: en,
	ja: ja
};
function i18n(Component) {
  const options = typeof Component === 'function' ? Component.options : Component;
  options.__i18n = options.__i18n || [];
  options.__i18n.push(JSON.stringify(__i18n));
}

/* script */
const __vue_script__ = script;

/* template */
var __vue_render__ = function() {
  var _vm = this;
  var _h = _vm.$createElement;
  var _c = _vm._self._c || _h;
  return _c(\\"p\\", [_vm._v(_vm._s(_vm.$t(\\"hello\\")))])
};
var __vue_staticRenderFns__ = [];
__vue_render__._withStripped = true;

  /* style */
  const __vue_inject_styles__ = undefined;
  /* scoped */
  const __vue_scope_id__ = undefined;
  /* module identifier */
  const __vue_module_identifier__ = undefined;
  /* functional template */
  const __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  
  /* style inject shadow dom */
  

  
  const __vue_component__ = normalizeComponent(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    false,
    undefined,
    undefined,
    undefined
  );
i18n(__vue_component__);

export default __vue_component__;
export { en, ja };
"
`;

exports[`yaml 1`] = `
"import { normalizeComponent } from 'vue-runtime-helpers';

//
//
//
//

var script = {
  name: 'Entry'
};

var data = { en:{ hello:\\"hello!\\" },
  ja:{ hello:\\"こんにちは！\\" } };

const __i18n = data;
var en = data.en;
var ja = data.ja;function i18n(Component) {
  const options = typeof Component === 'function' ? Component.options : Component;
  options.__i18n = options.__i18n || [];
  options.__i18n.push(JSON.stringify(__i18n));
}

/* script */
const __vue_script__ = script;

/* template */
var __vue_render__ = function() {
  var _vm = this;
  var _h = _vm.$createElement;
  var _c = _vm._self._c || _h;
  return _c(\\"p\\", [_vm._v(_vm._s(_vm.$t(\\"hello\\")))])
};
var __vue_staticRenderFns__ = [];
__vue_render__._withStripped = true;

  /* style */
  const __vue_inject_styles__ = undefined;
  /* scoped */
  const __vue_scope_id__ = undefined;
  /* module identifier */
  const __vue_module_identifier__ = undefined;
  /* functional template */
  const __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  
  /* style inject shadow dom */
  

  
  const __vue_component__ = normalizeComponent(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    false,
    undefined,
    undefined,
    undefined
  );
i18n(__vue_component__);

export default __vue_component__;
export { en, ja };
"
`;
